backend/routes/dashboardVisuals.js

```diff
@@ -0,0 +1,15 @@
+const express = require('express');
+const router = express.Router({ mergeParams: true });
+
+const {
+    getSalesByTime
+} = require('../controllers/saleController');
+
+// get authentication middleware
+const { protect, hasPermissions } = require('../middleware/auth');
+
+// add .get(protect, getSalesByTime) later
+router.route('/visual3')
+    .get(getSalesByTime)
```

Comment:
Add the protect middileware before the getSalesByTime

Reply:
Fixed!